src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java

```diff
@@ -158,10 +159,27 @@ public void generateNewPasswordForUser(String email) throws MessagingException,
         User user = userRepository.findOneByEmail(email).orElseThrow(
                 () -> new UserDoesNotExistException("User with this email does not exist"));
 
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
         user.setPassword(PasswordGenerator.generatePassword());
         String content = emailContent.createContent(user.getPassword(), "classpath:email/forgotPasswordEmail.html");
         emailSender.sendEmail(user, "Forgot Password", content);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+
+        userRepository.save(user);
+    }
+
+    public void editUserPassword(@RequestBody ChangePasswordDto changePasswordDto) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String currentUser = authentication.getName();
+
+        User user = userRepository.findOneByEmail(currentUser).orElseThrow(() ->
+                new NotLoggedInException("You are not logged in!"));
 
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        user.setPassword(passwordEncoder.encode(changePasswordDto.getNewPassword()));
```

Comment:
Think how to reuse the code, removing duplication in generateNewPasswordForUser() and editUserPassword()

Reply:
Are you talking about the BCryptPasswordEncoder?

Reply:
In general, for the repeating code.